packages/flutter/test/widgets/widget_inspector_test.dart

```diff
@@ -2331,6 +2331,41 @@ class TestWidgetInspectorService extends Object with WidgetInspectorService {
       );
     }, skip: isBrowser);
 
+    testWidgets('ext.flutter.inspector.structuredErrors', (WidgetTester tester) async {
+      List<Map<Object, Object>> flutterErrorEvents = service.getEventsDispatched('Flutter.Error');
+      expect(flutterErrorEvents, isEmpty);
+
+      final FlutterExceptionHandler oldHandler = FlutterError.onError;
+
+      try {
+        // enable
+        expect(await service.testBoolExtension(
+            'structuredErrors', <String, String>{'enabled': 'true'}),
+            equals('true'));
+
+        // create an error
+        FlutterError.reportError(FlutterErrorDetailsForRendering(
+          library: 'rendering library',
+          context: ErrorDescription('during layout'),
+          exception: StackTrace.current,
+        ));
+      } finally {
+        FlutterError.onError = oldHandler;
+      }
+
+      // validate that we received an error
+      flutterErrorEvents = service.getEventsDispatched('Flutter.Error');
+      expect(flutterErrorEvents, hasLength(1));
+
+      // validate the error contents
+      final Map<Object, Object> error = flutterErrorEvents.first;
+      expect(error['description'], 'Exception caught by rendering library');
+      expect(error['children'], isEmpty);
+
+      // validate that we received an error count
+      expect(error['errorsSinceReload'], 0);
+    });
```

Comment:
Thanks for adding a test for this! Can you also dispatch a second error (or the same error) and verify that the errorsSinceReload count increases?
You could also dispatch the event for a hot reload and verify the count is back to zero.

Reply:
👍 will do